elide-core/src/main/java/com/yahoo/elide/graphql/Environment.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.graphql;
+
+import com.google.common.collect.ImmutableList;
+import com.yahoo.elide.core.PersistentResource;
+import com.yahoo.elide.core.RequestScope;
+import graphql.language.Field;
+import graphql.schema.DataFetchingEnvironment;
+import graphql.schema.GraphQLType;
+
+import javax.ws.rs.WebApplicationException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.yahoo.elide.graphql.ModelBuilder.ARGUMENT_ID;
+
+/**
+ * Encapsulate the GraphQL request environment.
+ */
+public class Environment {
+    public static final List<Map<String, Object>> EMPTY_DATA = ImmutableList.of();
+
+    public final RequestScope requestScope;
+    public final Optional<String> id;
+    public final Object source;
+    public final PersistentResource parentResource;
+    public final GraphQLType parentType;
+    public final GraphQLType outputType;
+    public final Field field;
+    public final List<Map<String, Object>> data;
+    public final Optional<String> filters;
+
+    public Environment(DataFetchingEnvironment environment) {
```

Comment:
I'm wondering if we should have multiple classes - one for each operation type (FETCH, REPLACE, ADD, DELETE) that encapsulates the constraints and request state of each operation.